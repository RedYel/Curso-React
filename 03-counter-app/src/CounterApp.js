// FC
import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ( { value } ) => {

    const [ counter, setCounter ] = useState( 0 );
    //console.log(nombre, setNombre);

    const handleAdd = (e) => {
        //setCounter( counter + 1 );
        setCounter( (c) => c + 1 );
    }

    return (               
        <>
            <h1> CounterApp </h1>            
            <p> { counter } </p>
            {/*<button onClick={ (e) => {handleAdd(e)} }>+1</button>*/}
            <button onClick={ handleAdd }>+1</button>
        </>        
    );
    
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;