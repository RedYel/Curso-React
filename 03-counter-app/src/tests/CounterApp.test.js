
import React from 'react';
import { shallow } from 'enzyme'

import CounterApp from '../CounterApp';

describe('Pruebas en el CounterApp', () => {

    let wrapper = shallow( <CounterApp/>);

    beforeEach( () => {
        wrapper = shallow( <CounterApp/>);
    });

    test('Debe mostar counterApp correctamente', () => {
    
        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe mostrar un valor de 100 por defecto', () => {
      
        const wrapper = shallow( <CounterApp value={ 100 } />);
        
        const  counterText = wrapper.find('p').text().trim();
        
        expect( counterText ).toBe('100');
        //console.log(counterText)
    });
    
    test('Debe incrementar el contador', () => {
      
        wrapper.find('button').at(0).simulate('click');

        const  counterText = wrapper.find('p').text().trim();
        //console.log(counterText)
        expect( counterText ).toBe('11');
    });

    test('Debe decrementar el contador', () => {
      
        wrapper.find('button').at(2).simulate('click');
        const  counterText = wrapper.find('p').text().trim();
        //console.log(counterText)
        expect( counterText ).toBe('9');
    });
    
    test('Debe de colocar el valor por defecto con el btn reset', () => {
      
        const wrapper = shallow(<CounterApp value={ 105 }/>);

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');

        wrapper.find('button').at(1).simulate('click');

        const  counterText = wrapper.find('p').text().trim();
        //console.log(counterText)

        expect( counterText ).toBe( '105' );
    });
    

});
