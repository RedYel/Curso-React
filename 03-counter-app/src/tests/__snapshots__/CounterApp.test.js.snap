// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el CounterApp Debe mostar counterApp correctamente 1`] = `
<Fragment>
  <h1>
     CounterApp 
  </h1>
  <p>
     
    10
     
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
